src/mtype.h

```diff
@@ -131,6 +131,7 @@ enum m_flag {
     MF_GROUP_BASH,          // Monsters that can pile up against obstacles and add their strength together to break them.
     MF_SWARMS,              // Monsters that like to group together and form loose packs
     MF_GROUP_MORALE,        // Monsters that are more courageous when near friends
+    MF_CLIMBS,              // Monsters that can climb certain terrain and furniture
```

Comment:
You also need to add a line to `MonsterGenerator::init_flags()` to make this work.